thrust/testing/zip_function.cu

```diff
@@ -176,4 +176,23 @@ struct TestNestedZipFunction2
   }
 };
 SimpleUnitTest<TestNestedZipFunction2, type_list<int, float>> TestNestedZipFunctionInstance2;
-#endif // _CCCL_STD_VER
+
+#  if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
+template <class T>
+struct ProclaimReturnType
+{
+  void operator()()
+  {
+    auto fun =
+      thrust::make_zip_function(cuda::proclaim_return_type<thrust::tuple<int, int>>([] __device__(int a, int b) {
+        return thrust::make_tuple(a, b);
+      }));
```

Comment:
Important: I am unsure whether your specialization of `zip_function` works for `__host__` or `__host__ __device__` lambdas, and also depending whether the tarnsform iterator below is called from host or device code. Please extend the tests to cover those cases.